src/components/mainView/index.tsx

```diff
@@ -1,40 +1,88 @@
-import StudentInputModal from "components/inputModal";
-import StudentDetails from "components/mainView/StudentDetails";
 import StudentSummary from "components/mainView/StudentSummary";
-import { useState } from "react";
-import { Stack } from "react-bootstrap";
+import { useEffect, useState } from "react";
 import React from "react";
-import 'components/mainView/styles.css'
+import "components/mainView/styles.css";
+import { IStudentRaw, IStudentState } from "state/ducks/students/types";
+import Header from "components/mainView/header";
+import { ActionType } from "typesafe-actions";
+import { fetchStudents } from "state/ducks/students/actions";
+import { subjects, ISummaryData } from "utils";
+import InputModalContainer from "containers/inputModalContainer";
+import StudentDetailsContainer from "containers/studentDetailsContainer";
+interface IProps extends IStudentState {
+  fetchStudents: () => ActionType<typeof fetchStudents>;
+  data: IStudentRaw[];
+}
+
+const MainView: React.FC<IProps> = ({ fetchStudents, data }) => {
+  const [show, setShow] = useState(false);
+  const [studentData, setStudentData] = useState<IStudentRaw>();
+  const [dashboardData, setDashboardData] = useState<ISummaryData>();
+  useEffect(() => {
+    fetchStudents();
+  }, [fetchStudents]);
+
+  useEffect(() => {
+    summaryData();
+  }, [data]);
+
+  const summaryData = () => {
+    let result = {} as ISummaryData;
+    const sortedGrades = data.map((itx) => itx.grade).sort();
+
+    result.topGrade = sortedGrades[0];
+    result.minGrade = sortedGrades[sortedGrades.length - 1];
 
-function MainView() {
-  const [show, setShow] = useState(false)
+    //initialize hash-map
+    const passed = new Map();
```

Comment:
why do you need to use hash map?

Reply:
for better performance and a better alternate to using a multi-dimensional array to keep track of the passed/failed subjects.